lib/static/new-ui/components/ToolbarOverlay/index.tsx

```diff
@@ -0,0 +1,80 @@
+import {Icon} from '@gravity-ui/uikit';
+import {Grip} from '@gravity-ui/icons';
+import classNames from 'classnames';
+import React, {ReactNode, useState} from 'react';
+import {createPortal} from 'react-dom';
+
+import styles from './index.module.css';
+import {Point} from '@/static/new-ui/types';
+
+interface ToolbarOverlayProps {
+    isVisible: boolean | null;
+    className?: string;
+    children: ReactNode;
+    draggable?: {
+        position: Point;
+        onPositionChange: (position: Point) => void;
+    }
+}
+
+export function ToolbarOverlay(props: ToolbarOverlayProps): ReactNode {
+    const [dragging, setDragging] = useState(false);
+    const [offset, setOffset] = useState({x: 0, y: 0});
+
+    const handleMouseDown = (e: React.MouseEvent): void => {
+        if (!props.draggable) {
+            return;
+        }
+
+        setDragging(true);
+        setOffset({
+            x: e.clientX - props.draggable.position.x,
+            y: e.clientY - props.draggable.position.y
+        });
+    };
+
+    const handleMouseMove = (e: MouseEvent): void => {
+        if (!dragging || !props.draggable) {
+            return;
+        }
+
+        const newPosition = {
+            x: e.clientX - offset.x,
+            y: e.clientY - offset.y
+        };
+        props.draggable.onPositionChange(newPosition);
+    };
+
+    const handleMouseUp = (): void => {
+        setDragging(false);
+    };
```

Comment:
What exactly will we gain from `useCallback` here? I think it will do more harm than good in this case. (https://kentcdodds.com/blog/usememo-and-usecallback)

> Did you check if it actually removes event listeners in Event Listeners tab in devtools?

yes it does